nidmviewerfsl/lib/slicerTools.py

```diff
@@ -210,38 +102,63 @@ def getSliceImageFromNifti(tempDir, outputName):
     process.wait()
 
 
-def generateSliceImage_SPM(exc_set):
+def generateSliceImage(exc_set, SPMorFSL):
 
     tempFolder = 'temp_NIDM_viewer' + str(random.randint(0, 999999))
     os.mkdir(tempFolder)
     FSLDIR = os.environ['FSLDIR']
 
-    # Find the template. If we can't find an appropriate template scaling will
-    # be required later.
-    if nifDim(exc_set, 'pix') == 1:
-        template = os.path.join(FSLDIR, 'data', 'standard',
-                                'MNI152_T1_1mm_brain.nii.gz')
-        scalefactor = 1
-    elif nifDim(exc_set, 'pix') == 2:
-        template = os.path.join(FSLDIR, 'data', 'standard',
-                                'MNI152_T1_2mm_brain.nii.gz')
-        scalefactor = 1
+    # Make a copy of the original name of the excursion set.
+    o_exc_set = exc_set
+
+    # If we are looking at FSL data use the FSL template.
+    if SPMorFSL == 'FSL':
+        if nifDim(exc_set, 'pix') == 1:
+            template = os.path.join(FSLDIR, 'data', 'standard',
+                                    'MNI152_T1_1mm_brain.nii.gz')
+        else:
+            template = os.path.join(FSLDIR, 'data', 'standard',
+                                    'MNI152_T1_2mm_brain.nii.gz')
     else:
-        template = os.path.join(FSLDIR, 'data', 'standard',
-                                'MNI152_T1_2mm_brain.nii.gz')
-        scalefactor = 1/nifDim(exc_set, 'pix')
-
-    # Check which is bigger and resize if necessary
-    resizeSPMtoFSL(exc_set, template, scalefactor, tempFolder)
-    resized_exc_set = os.path.join(tempFolder, 'resizedNifti.nii.gz')
+        #  NaN values.
+        n = nib.load(exc_set)
+        d = n.get_data()
+        exc_set_nonan = nib.Nifti1Image(np.nan_to_num(d),
+                                        n.affine,
+                                        header=n.header)
+
+        # Save the result.
+        nib.save(exc_set_nonan, os.path.join(tempFolder,
+                                             'excset_nonan.nii.gz'))
+        exc_set = os.path.join(tempFolder, 'excset_nonan.nii.gz')
+
+        # Use the SPM template.
+        template = os.path.join(
+            os.path.split(
+                os.path.split(
+                    os.path.split(os.path.realpath(__file__))[0])[0])[0],
+            'templates', 'T1_skullStripped.nii')
+
+    if nifDim(exc_set, 'x') != nifDim(template, 'x'):
+        # Check which is bigger and resize if necessary
```

Comment:
Can you briefly explain why you are reslicing one way around or the other around depending on size?

Reply:
When I first coded this resampling the smaller image to the larger image sometimes affected the image in strange ways. Blank horizontal lines appeared and made the image appear sketchy.

Reply:
Okay... Let's be consistent. Can you remove this condition on size and if we have the black horizontal lines we can investigate why this is happening? (We might have to adapt the order of the spline interpolation, i.e. probably use 0 for nearest neighbors).